reactnative-todo/ios/reactnativetodo/FRAuthSampleStructs.swift

```diff
@@ -0,0 +1,250 @@
+      self.inputNames = [thisCallback.inputName!]
+    }
+
+    if let thisCallback = callback as? AbstractValidatedCallback {
+      if let policyDictionary = thisCallback.policies,
+        let policiesJSON = try? policyDictionary.toJson() {
+        
+        let jsonData = Data(policiesJSON.utf8)
+
+        do {
+          self.policies = try JSONDecoder().decode(RawPolicies.self, from: jsonData)
+        }
+        catch {
+          print(error)
+        }
+      }
+      if let failedPolicies = thisCallback.failedPolicies {
+        self.failedPolicies = []
+        for failedPolicy in failedPolicies {
+          var paramsDictionary = [String: FlexibleType]()
+          if let params = failedPolicy.params {
+            let newDictionary = params.mapValues { value -> FlexibleType in
+              if let str = value as? String {
+                return FlexibleType(str, originalType: .String)
+              } else if let str = value as? Int {
+                return FlexibleType(String(str), originalType: .Int)
+              } else if let str = value as? Double {
+                return FlexibleType(String(str), originalType: .Double)
+              } else if let str = value as? Bool {
+                return FlexibleType(String(str), originalType: .Bool)
+              } else {
+                return FlexibleType("", originalType: .String)
+              }
+            }
+            paramsDictionary = newDictionary
+          }
+          self.failedPolicies?.append(
+            RawFailedPolicies(
+              propertyName: self.prompt,
+              params: paramsDictionary,
+              policyRequirement: failedPolicy.policyRequirement,
+              failedDescription: failedPolicy.failedDescription()
+            )
+          )
+        }
+      }
+    }
+
+    if let jsonData = try? JSONSerialization.data(withJSONObject: callback.response, options: .prettyPrinted),
+      let jsonString = String(data: jsonData, encoding: .utf8) {
+      self.response = jsonString
+    } else {
+      self.response = ""
+    }
+  }
+}
+
+public struct Response: Codable {
+  var authId: String?
+  var callbacks: [RawCallback]?
+  var status: Int?
+}
+
+public struct RawCallback: Codable {
+  var type: String?
+  var input: [RawInput]?
+  var _id: Int?
+}
+
+public struct RawPolicies: Codable {
+  var name: String?
+  var policyRequirements: [String]?
+  var policies: [Policy]?
+}
+
+public struct RawFailedPolicies: Codable {
+  var propertyName: String?
+  var params: [String: FlexibleType]?
+  var policyRequirement: String?
+  var failedDescription: String?
+}
+
+public struct Policy: Codable {
+  var policyId: String?
+  var policyRequirements: [String]?
+  var params: [String: FlexibleType]?
+}
+
+public struct RawInput: Codable {
+  var name: String
+  var value: FlexibleType?
+}
+
+public enum ResponseType {
+  case String
+  case Int
+  case Double
+  case Bool
+  case TypeMismatch
+  case NotSet
+}
```

Comment:
In swift enum cases naming convention is lowerCamelCase (I think before Swift 3 it was UpperCamelCase)

Reply:
Good point